abra/data.py

```diff
@@ -1,22 +1,178 @@
+        start_time = ''
+        end_time = ''
+        flag = False
+        events = []
+        events_dict = {}
+        timestamps_list = []
+        messages_dict = {}
+        trial_markers = {"start":[], "end":[]}
+        pupil_size_list = []
+        movement_list = [[],[]]
+        #Default Mode
+        if mode == "d":
+            for num, line in enumerate(f, 1):
+                elements = line.split()
+                if line.startswith('START'):
+                    start_time = elements[1]
+                    trial_markers["start"].append(elements[1])
+                    end_time = ''   #to end "END message input"
+                    timestamps_list.append(elements[1])
+
+                #to only get END messages
+                #adds to trial_markers, messages_dict, timestamps_list
+                if line.startswith('END'):
+                    end_time = elements[1]
+                    trial_markers["end"].append(elements[1])
+                    messages_dict[elements[1]] = elements[2:]
+                    timestamps_list.append(elements[1])
+                    flag = False
+                    start_time = ''
+
+                if start_time:
+                    if line.startswith(start_time):
+                        flag = True
+                # check for start
+                if flag == True:
+                    # will get pupil size, timestamps, and movements
+                    if is_number(elements[0]):
+                        timestamps_list.append(elements[0])
+                        pupil_size_list.append(elements[1])
+                        movement_list[0].append(elements[2]) # x-axis
+                        movement_list[1].append(elements[3]) # y-axis
+                    # Gets all messages between START and END
+                    elif elements[0] == "MSG":
+                        timestamps_list.append(elements[1])
+                        messages_dict[elements[1]] = elements[2:]
+                    # Gets all events between START and END
+                    elif not is_number(elements[1]):
+                        events.append(elements[0])
+                        timestamps_list.append(elements[2])
+                        # check if event already exist
+                        if str(elements[0] + " " + elements[1]) not in events_dict:
+                            events_dict[str(elements[0] + " " + elements[1])] = elements[2:]
+                        else:
+                            events_dict[str(elements[0] + " " + elements[1])].append(elements[2:])
+
+                #finds all the END messages that is outputed
+                if end_time:
+                    if line.startswith('MSG'):
+                        if elements[1] in messages_dict:
+                            messages_dict[elements[1]].append(elements[2:])
+                        else:
+                            messages_dict[elements[1]] = elements[2:]
+    # User Defined Mode
+    elif mode == "u":
```

Comment:
@AkimaCon @andylikescodes 
`elif` block needs to be indented one tab further, tests didn't run properly after they get here due to incorrect tab indentation. I will push this code and fix the indentation so no need to change anything on your end. Branch will be closed after merge.